### DISCUSSION TITLE
Docker image tagged versions

### DISCUSSION THREAD

**Discussion by mddeff** [inventree/InvenTree#3320] [Q&A] [answered]
Hey,

First: Amazing project! I was blown away when my buddy showed me this last Friday!  Thank you to all the contributors of this project for making something like this accessible to mere mortals.

My question comes as I'm trying to write a Helm chart for this app.  I based it off of the [production docker compose file](https://github.com/inventree/InvenTree/blob/master/docker/production/docker-compose.yml).  In there, I noticed it's pointing at the docker **stable** tag, which, based off of the GitHub releases, is version `0.7.4`. But **hub.docker.com** tells a different tale:

![Screenshot from 2022-07-11 08-58-17](https://user-images.githubusercontent.com/16808640/178269634-0ccdb37d-1677-4d80-afa1-b612bc43c916.png)

While the `0.7.4` tag and the `stable` tag were pushed at the same time, that seems to be where the similarities end.  The hashes are different, and the `0.7.4` tag is almost 50MB larger.  It's usually best practice to point at an image tag in the helm chart that doesnt change, so I'm trying to avoid `stable`.  Any insight to the differences?  Thanks!

(If I missed something obvious in the repo/docs that explains this, my apologies.)

**Comment by SchrodingersGat** [comment, marked as answer]
@mddeff thanks for your kind words 👍 

You are right that `0.7.4` and `stable` should be the same image. That is how it is intended. I was not aware that the images were building differently

From looking through the image digests for [stable](https://hub.docker.com/layers/inventree/inventree/inventree/stable/images/sha256-6ef2d7cc5a05e13eb005e73d7319d31d73c2c3a10857a0cfb84688553ab86850?context=explore) and [0.7.4](https://hub.docker.com/layers/inventree/inventree/inventree/0.7.4/images/sha256-fcc9e8277c19c7e5c0235cae0dc042dce6d4a4bf484498dfd28e37b392454800?context=explore) it appears they are mostly the same, except for this step:

![image](https://user-images.githubusercontent.com/10080325/178372335-cc699ed5-42bf-4273-8c25-5df191941958.png)

The docker build behaves differently if it is running from a git `tag` or a `branch`.  So functionally the images are the same but one contains some more git history.

For now, you can continue under the assumption that `stable` always points to the latest release.

Please note that for the next major release (`0.8.0`) we have changed the way that the docker build works, so this particular problem will no longer be an issue

I hope that you enjoy using our software, and thanks for pointing this out.

**Comment by matmair** [comment]
@mddeff were you successful in creating a helm chart? Might be an interesting deployment method.
